tools/mage/js.go

```diff
@@ -76,7 +77,32 @@ func (js Js) runWebpack(config string, args ...string) error {
 }
 
 func (js Js) runEslint(args ...string) error {
-	return js.runYarnCommandV("eslint", append([]string{"--color", "--no-ignore"}, args...)...)
+	return js.runYarnCommand("eslint", append([]string{"--color", "--no-ignore"}, args...)...)
+}
+
+func (js Js) waitOn() error {
+	u, err := url.Parse(js.getURL())
+	if err != nil {
+		return err
+	}
+	return js.runYarnCommand("wait-on", append([]string{
+		fmt.Sprintf("--timeout=%d", 120000),
+		fmt.Sprintf("--interval=%d", 1000),
+		fmt.Sprintf("http-get://%s/oauth", u.Host),
+	})...)
+}
+
+func (js Js) runCypress(command string, args ...string) error {
+	mg.Deps(js.waitOn)
+
+	return js.runYarnCommand("cypress", append([]string{command, fmt.Sprintf("--config-file=%s", filepath.Join("config", "cypress.json")), "--config", fmt.Sprintf("baseUrl=%s", js.getURL())}, args...)...)
+}
+
+func (js Js) getURL() string {
```

Comment:
1. Avoid `getX()`, just write `X()` - this is a convention in Go
2. `URL` is too ambiguous
```suggestion
func (js Js) frontendURL() string {
```
Or maybe you have a better suggestion?